src/main/java/com/nutomic/syncthingandroid/util/ConfigXml.java

```diff
@@ -63,19 +67,75 @@ public ConfigXml(Context context) throws OpenConfigException {
             generateKeysConfig(context);
         }
 
-        try {
-            DocumentBuilder db = DocumentBuilderFactory.newInstance().newDocumentBuilder();
-            mConfig = db.parse(mConfigFile);
-        } catch (SAXException | ParserConfigurationException | IOException e) {
-            throw new OpenConfigException();
-        }
+	readConfig();
 
         if (isFirstStart) {
             changeLocalDeviceName();
             changeDefaultFolder();
         }
     }
 
+    private void readConfig() {
+	if (!mConfigFile.canRead() && !checkPermConfigFile()) {
+	    throw new OpenConfigException();
+	}
+        try {
+            DocumentBuilder db = DocumentBuilderFactory.newInstance().newDocumentBuilder();
+	    Log.d(TAG,"Trying to read '" + mConfigFile + "'");
+            mConfig = db.parse(mConfigFile);
+        } catch (SAXException | ParserConfigurationException | IOException e) {
+	  Log.w(TAG,"Cannot read '" + mConfigFile + "'",e);
+	  throw new OpenConfigException();
+	}
+    }
+
+    private boolean checkPermConfigFile() {
+	// Since either readConfig or saveChanges called this method, the permissions of config.xml got messed up. 
+	// For now, the only way to do so is to run Syncthing as root, because the config file resides in the application's data directory.
+	// That directory is only writeable by this application, and root of course. Syncthing will save it's config as root with 0600.
+	// We can safely assume that root magic is somehow available.
+	// Be paranoid :) and check if su binary is available.
+	if (!Shell.SU.available()) {
+	    Log.e(TAG,"Su is not available. Cannot fix permssions ");
+	    return false;
+	}
+
+	try {
+		ApplicationInfo appInfo = mContext.getPackageManager().getApplicationInfo(mContext.getPackageName(),0);
+		Log.d(TAG,"Uid of '" + mContext.getPackageName() + "' is " + appInfo.uid);
+		try {
+		    Process fixPerm = Runtime.getRuntime().exec("su");
+		    DataOutputStream fixPermOut = new DataOutputStream(fixPerm.getOutputStream());
+		    String cmd = "chown " + appInfo.uid + ":" + appInfo.uid + " " + mConfigFile + "\n";
```

Comment:
I didn't know about appInfo.uid, that's very useful. We should probably also use that for SettingsActivity.ChownFilesRunnable (that seems to be broken at the moment, so the app doesnt start after disabling root). Feel free to add a commit for that.

Reply:
Can you pls give me a short explanation why this method is necessary?
I do not get the point here. :(

FYI, the chmod of the Syncthing binary seems to be broken, too. It is blocked by SELinux: 
09-19 23:36:31.720 18241 18241 W chmod   : type=1400 audit(0.0:1122): avc: denied { setattr } for name="libsyncthing.so" dev="dm-0" ino=426157 scontext=u:r:untrusted_app:s0:c512,c768 tcontext=u:object_r:apk_data_file:s0 tclass=file permissive=0

Reply:
To make sure database files etc are owned by the app (not root), when the root setting is disabled.

Reply:
Regarding chmod, I'm not quite sure why we need that. @Zillode

Reply:
Are you sure?
I disabled root manually by editing the shared_prefs.xml. O:-) This is somehow similar to removing that part of the code. I now cannot start Syncthing, because it cannot read the database.

Reply:
That's exactly why this method exists. When root is disabled in the settings, it chowns all files in the app folder to make sure they are readable.